pkg/pillar/types/conntest.go

```diff
@@ -40,18 +51,25 @@ func (trPtr *TestResults) HasError() bool {
 	return trPtr.LastFailed.After(trPtr.LastSucceeded)
 }
 
+// HasWarning returns true if test succeeded but there is a warning reported.
+func (trPtr *TestResults) HasWarning() bool {
+	return !trPtr.HasError() && trPtr.LastWarning != ""
```

Comment:
Shouldn't the same logic be used here as used in the `HasError()`? Something like 
```
return trPtr.LastWarned.After(trPtr.LastSucceeded) && ...
```

Reply:
I don't think that is necessary. Warning is used when connectivity test is successful but there is some potential issue with the config that the user should be aware of. So we continue using the `LastSucceeded` timestamp in this case.